test/ranking/ranking.test.ts

```diff
@@ -30,4 +31,64 @@ describe('ranking', () => {
       assert.deepEqual(group.items, []);
     });
   });
+
+  describe('comparatorFactory', () => {
+    it('should create a comparator that returns a score difference when passed an orderBy array', () => {
+      const specM1 = SpecQueryModel.build(
+        {
+          mark: Mark.POINT,
+          encodings: [
+            {channel: Channel.X, field: 'Q', type: Type.QUANTITATIVE},
+            {channel: Channel.Y, field: 'Q1', type: Type.QUANTITATIVE},
+          ]
+        },
+        schema,
+        DEFAULT_QUERY_CONFIG
+      );
+
+      const specM2 = SpecQueryModel.build(
+        {
+          mark: Mark.BAR,
+          encodings: [
+            {channel: Channel.X, field: 'N', type: Type.NOMINAL},
+            {channel: Channel.Y, field: 'Q', type: Type.QUANTITATIVE},
+          ]
+        },
+        schema,
+        DEFAULT_QUERY_CONFIG
+      );
+
+      const comparator = comparatorFactory(['aggregationQuality', 'effectiveness'], schema, DEFAULT_QUERY_CONFIG);
+      assert.isNumber(comparator(specM1, specM2));
```

Comment:
This have two cases covered

> a) (aggregateQuality(left)  >  aggregateQuality(right)) 

Try  QxQ > MEAN(Q)xMEAN(Q)

> b) (aggregateQuality(left)  ===  aggregateQuality(right) && effectiveness(left)  >=  effectiveness(right))

Try something that should be plot with line then copy it and change to point.  Than Line > Point and the data is the same.  

You can reuse this example for the case that we input just `'aggregationQuality'` like below.